src/detect-flowbits.c

```diff
@@ -340,7 +358,10 @@ int DetectFlowbitSetup (DetectEngineCtx *de_ctx, Signature *s, const char *rawst
         case DETECT_FLOWBITS_CMD_ISNOTSET:
         case DETECT_FLOWBITS_CMD_ISSET:
             /* checks, so packet list */
-            SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_MATCH);
+            if (on_data)
+                SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_PMATCH);
```

Comment:
this won't work and will trigger https://github.com/OISF/suricata/blob/master/src/detect-engine-content-inspection.c#L651

Reply:
the PMATCH list is only for matches that can be evaluated by `DetectEngineContentInspection`

Reply:
gdb will tell you `DetectFlowbitMatchIsnotset` will not be called